ros2/node/nns_rclcpp_subscriber.cc

```diff
@@ -247,7 +298,10 @@ void NnsRclCppSubscriber::loop ()
   );
 
   while (true) {
-    rclcpp::spin_some (this->node_);
+    if (!this->is_dummy)
+      rclcpp::spin_some (this->node_);
+    else
+      this->fetchDataFromBag ();
```

Comment:
Who's using the lock ```lk``` of line 307?

Reply:
That is a guard for the finalize method (at line 187) that would run on another thread.

Reply:
That is a constructor. Don't you need to call ```std::lock(lk);``` in the scope of ```lk```?

Reply:
@myungjoo IMHO, '''unique_lock''' does not require invoking any '''std::lock''' method families. It accquires the given mutex, i.e., '''this->g_m''' on construction and releases that mutex on destruction.

Reply:
IIRC, `std::unique_lock()` basically works like `std::lock_guard()`.
However, if `std::defer_lock` option is used when calling `std::unique_lock()`, `std::lock()` is explicitly necessary to lock the critical section.

Reply:
Ah.. this locks when declared and unlocks when it goes out of the block. Great!